src/main/java/org/cactoos/scalar/ScalarOf.java

```diff
@@ -23,32 +23,61 @@
  */
 package org.cactoos.scalar;
 
+import org.cactoos.Func;
+import org.cactoos.Proc;
 import org.cactoos.Scalar;
+import org.cactoos.func.FuncOf;
 
 /**
  * ScalarOf.
  *
+ * @param <X> Element type
  * @param <T> Element type
  * @since 0.4
  */
-public final class ScalarOf<T> implements Scalar<T> {
+public final class ScalarOf<X, T> extends ScalarEnvelope<T> {
+    /**
+     * Ctor.
+     * @param runnable Encapsulated proc
+     * @param result Result to return
+     * @since 0.32
+     */
+    public ScalarOf(final Runnable runnable, final T result) {
+        this(
+            () -> {
+                runnable.run();
+                return result;
+            }
+        );
+    }
 
     /**
-     * The scalar.
+     * Ctor.
+     * @param proc Encapsulated proc
+     * @param ipt Input
+     * @param result Result to return
+     * @since 0.41
      */
-    private final Scalar<T> origin;
+    public ScalarOf(final Proc<X> proc, final X ipt, final T result) {
+        this(new FuncOf<>(proc, result), ipt);
+    }
 
     /**
      * Ctor.
-     *
-     * @param origin The scalar
+     * @param fnc Encapsulated func
+     * @param ipt Input
+     * @since 0.41
      */
-    public ScalarOf(final Scalar<T> origin) {
-        this.origin = origin;
+    public ScalarOf(final Func<X, T> fnc, final X ipt) {
```

Comment:
@fabriciofx `Func` & `Proc` parameters should follow PECS (not sure about `Scalar``).

Reply:
@andreoss done!